core/src/main/java/io/stargate/core/metrics/api/Metrics.java

```diff
@@ -27,4 +28,12 @@
    * automatically added when registering new metrics.
    */
   MetricRegistry getRegistry(String prefix);
+
+  /**
+   * Returns a Micrometer {@link io.micrometer.core.instrument.MeterRegistry} that can be used to
+   * add micrometer styled metrics with tags.
+   *
+   * <p>Note that this registry will warp all the metrics reported via {@link #getRegistry()}.
+   */
+  PrometheusMeterRegistry getMeterRegistry();
```

Comment:
We used to have a method with a prefix for getting the registry in "write" mode (cf. line 30). How are modules expected to register micrometer metrics to avoid collision?

Reply:
Was not part of this PR imo, will think on this later on.. But can be done:
* create new composite registry that contains the Prometheus one and simply change naming convention to always add a prefix..